helper/impls/helper-impl-110/src/main/java/com/linkedin/avroutil1/compatibility/avro110/FieldBuilder110.java

```diff
@@ -92,6 +93,49 @@ public FieldBuilder setOrder(Order order) {
     return this;
   }
 
+  @Override
+  public FieldBuilder addProp(String propName, String jsonLiteral) {
+    if (propName == null || jsonLiteral == null) {
+      throw new IllegalArgumentException("Function input parameters cannot be null.");
+    }
+    if (_props == null) {
+      _props = new HashMap<>();
+    }
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      _props.put(propName, objectMapper.readTree(jsonLiteral));
+      return this;
+    } catch (JsonProcessingException e) {
+      throw new IllegalArgumentException("Failed to parse serialized json object: " + jsonLiteral, e);
+    }
+  }
+
+  @Override
+  public FieldBuilder addProps(Map<String, String> propNameToJsonObjectMap) {
+    if (propNameToJsonObjectMap == null) {
+      throw new IllegalArgumentException("Function input parameters cannot be null.");
+    }
+    for (Map.Entry<String, String> entry : propNameToJsonObjectMap.entrySet()) {
+      try {
+        addProp(entry.getKey(), entry.getValue());
+      } catch (IllegalArgumentException e) {
+        throw new IllegalArgumentException(
+            "Issue with adding prop with key: " + entry.getKey() + " and value: " + entry.getValue(), e);
+      }
+    }
+    return this;
+  }
+
+
+  @Override
+  public FieldBuilder removeProp(String propName) {
+    if (propName == null || _props == null || !_props.containsKey(propName)) {
+      throw new IllegalArgumentException("Cannot remove prop that doesn't exist: " + propName);
+    }
+    _props.remove(propName);
+    return null;
```

Comment:
this is a potential NPE if used in succession.
Like 
`fieldBuilder.removeProp("A").removeProp("B")`

Reply:
Doh! Good catch